Microsoft.Toolkit.Uwp.UI/Triggers/ControlSizeTrigger.cs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Triggers
+{
+    /// <summary>
+    /// A conditional state trigger that functions
+    /// based on the target control's width or height.
+    /// </summary>
+    public class ControlSizeTrigger : StateTriggerBase
```

Comment:
We could add a couple of unit tests for this, eh?

Reply:
I'm having a bit of trouble testing this trigger. Here's what I have so far, but the assert isn't working. It seems like my setup isn't done properly. Any ideas?
```csharp
        public async Task WidthTriggerTest()
        {
            await App.DispatcherQueue.EnqueueAsync(() =>
            {
                var rootGrid = new Grid()
                {
                    Height = 1000,
                    Width = 1000
                };
                var grid = new Grid()
                {
                    Height = 100,
                    Width = 400
                };
                rootGrid.Children.Add(grid);

                var groups = VisualStateManager.GetVisualStateGroups(rootGrid);
                var stateGroup = new VisualStateGroup();
                var controlState = new VisualState();
                var testState = new VisualState();
                var trigger = new ControlSizeTrigger();
                trigger.TargetElement = grid;
                testState.StateTriggers.Add(trigger);
                stateGroup.States.Add(controlState);
                stateGroup.States.Add(testState);
                groups.Add(stateGroup);

                trigger.MaxWidth = 500;
                trigger.MinWidth = 400;
                Assert.IsTrue(stateGroup.CurrentState == testState);
            });
        }
```

Reply:
Found a workaround. Unit test added.